src/modules/core/components/popover/popover.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  type CSSProperties,
+  useState,
+  useEffect,
+  useRef,
+  forwardRef,
+} from 'react';
+import { createPortal } from 'react-dom';
+import { Transition, type TransitionStatus } from 'react-transition-group';
+import { useEventListener, useOnClickOutside } from 'usehooks-ts';
+// hooks
+import { useCombinedRefs } from '@/modules/core/hooks/use-combined-refs';
+
+import type { PopoverProps } from './popover.interface';
+import styles from './popover.module.scss';
+
+const transitionStyles: Record<TransitionStatus, CSSProperties> = {
+  entering: { opacity: 1, transform: 'scale(1)' },
+  entered: { opacity: 1, transform: 'scale(1)' },
+  exiting: { opacity: 0, transform: 'scale(0)' },
+  exited: { opacity: 0, transform: 'scale(0)' },
+  unmounted: {},
+};
+
+const IN_OUT_ANIMATION_DURATION = 300;
+
+const Popover = forwardRef<HTMLDivElement, PopoverProps>(
+  (
+    {
+      isOpen,
+      anchorElement,
+      anchorOrigin,
+      followAnchorWidth,
+      onRequestClose,
+      requestCloseOnOutsideClick = true,
+      children,
+    },
+    ref
+  ) => {
+    // state
+    const [rootStyle, setRootStyle] = useState<CSSProperties>({});
+    // refs
+    const rootRef = useRef<HTMLDivElement>(null);
+    const combinedRefs = useCombinedRefs<HTMLDivElement>(ref, rootRef);
+
+    useEffect(() => {
+      if (isOpen && anchorElement) {
+        const { top, bottom, left, right, height, width } =
+          anchorElement.getBoundingClientRect();
+
+        const verticalStyle: CSSProperties =
+          anchorOrigin?.vertical === 'top'
+            ? {
+                bottom: window.innerHeight - bottom + height,
+              }
+            : {
+                top: top + height,
+              };
+        const horizontalStyle: CSSProperties =
+          anchorOrigin?.horizontal === 'right'
+            ? {
+                right: window.innerWidth - right,
+              }
+            : {
+                left,
+              };
+
+        setRootStyle({
+          ...verticalStyle,
+          ...horizontalStyle,
+          ...(followAnchorWidth && {
+            width,
+            maxWidth: width,
+          }),
+        });
+      }
+    }, [anchorElement, isOpen, anchorOrigin, followAnchorWidth]);
+
+    useOnClickOutside(combinedRefs, () => {
```

Comment:
To be honest for me useOnclickOutside and useCombinedRefs looks a bit of complex that it should be. In my opinion it would be better to use some existing library that will do all this stuff for us, as the popover is very typical element.

Reply:
useOnClickOutside is from `usehooks-ts` library and I haven't found good implementation for `useCombinedRefs`.

Reply:
Okay.